cscs-checks/microbenchmarks/stream/stream.py

```diff
@@ -103,24 +100,23 @@ def __init__(self):
                 'daint:mc': {'triad': (119000, -0.05, None, 'MB/s')},
                 'dom:gpu': {'triad': (59500, -0.05, None, 'MB/s')},
                 'dom:mc': {'triad': (119000, -0.05, None, 'MB/s')},
-                '*': {'triad': (0.0, None, None, 'MB/s')},
             },
             'PrgEnv-pgi': {
                 'daint:gpu': {'triad': (44500, -0.05, None, 'MB/s')},
                 'daint:mc': {'triad': (88500, -0.05, None, 'MB/s')},
                 'dom:gpu': {'triad': (44500, -0.05, None, 'MB/s')},
                 'dom:mc': {'triad': (88500, -0.05, None, 'MB/s')},
-                '*': {'triad': (0.0, None, None, 'MB/s')},
             }
         }
         self.tags = {'production', 'craype'}
         self.maintainers = ['RS', 'SK']
 
-    def setup(self, partition, environ, **job_opts):
+    @rfm.run_before('compile')
+    def setup_per_partition_and_env(self):
```

Comment:
```python
@rfm.run_before('run')
def prepare_run(self):
```

Reply:
`self.build_system.cflags` is also set in the function, so it needs to be `@rfm.run_before('compile')`, right? Should I change the name to `prepare_compile` or split in more functions?

Reply:
Either split it or call it `prepare_test` and attach it after `setup`.